### PR TITLE
fetch npd from github releases

### PR DESCRIPTION
**Pull request by BenTheElder**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Stops depending on NPD being pushed to gs://kubernetes-release (the bucket currently behind dl.k8s.io, still in google.com instead of kubernetes.io GCP) by a Googler that happens to still have direct push access. Fetches from github releases instead.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

/sig testing node

### CODE REVIEW

File: cluster/gce/gci/configure.sh [hunk 1]
```diff
@@ -296,7 +296,7 @@ function install-node-problem-detector {
   fi
 
   echo "Downloading ${npd_tar}."
-  local -r npd_release_path="${NODE_PROBLEM_DETECTOR_RELEASE_PATH:-https://storage.googleapis.com/kubernetes-release}"
+  local -r npd_release_path="${NODE_PROBLEM_DETECTOR_RELEASE_PATH:-"https://github.com/kubernetes/node-problem-detector/releases/download/${npd_version}"}"
   download-or-bust "${npd_hash}" "${npd_release_path}/node-problem-detector/${npd_tar}"
```
Comment (wangzhen127) [inline cluster/gce/gci/configure.sh:300]: `download-or-bust "${npd_hash}" "${npd_release_path}/${npd_tar}"`

Example: https://github.com/kubernetes/node-problem-detector/releases/download/v0.8.16/node-problem-detector-v0.8.16-linux_amd64.tar.gz

**Comment by wangzhen127** [comment]
/lgtm cancel

Just a nit for the URL

**Comment by hakman** [comment]
@BenTheElder artifact promotion may also be a good idea in the future (similar to https://github.com/kubernetes/k8s.io/pull/6368).

**Comment by BenTheElder** [comment]
@hakman I don't believe the project should keep hosting subproject binaries ourselves. It's an unnecessary expense and infra. Discussion in #sig-k8s-infra

File: cluster/gce/gci/configure.sh (hunk 1 above)
Comment (BenTheElder) [reply cluster/gce/gci/configure.sh:300]: That would break users currently setting the existing capital env?

Comment (BenTheElder) [reply cluster/gce/gci/configure.sh:300]: I guess we're forced to

**Comment by BenTheElder** [comment]
Users will have to switch anyhow as it won't be gs:/:kubernetes-release and NPD has been publishing binaries to github already.

File: cluster/gce/gci/configure.sh (hunk 1 above)
Comment (wangzhen127) [reply cluster/gce/gci/configure.sh:300]: We are only making this change in master branch, right? Existing customers would not be impacted.

Comment (BenTheElder) [reply cluster/gce/gci/configure.sh:300]: I guess nothing is currently testing this aspect either since CI passed ..

Comment (BenTheElder) [reply cluster/gce/gci/configure.sh:300]: Nothing in the project is using this override https://cs.k8s.io/?q=NODE_PROBLEM_DETECTOR_RELEASE_PATH
And we don't support external users for these scripts (officially deprecated for years now), so it's OK to break.

Comment (BenTheElder) [reply cluster/gce/gci/configure.sh]: done.

File: cluster/gce/gci/configure.sh [hunk 2]
```diff
@@ -296,8 +296,8 @@ function install-node-problem-detector {
   fi
 
   echo "Downloading ${npd_tar}."
-  local -r npd_release_path="${NODE_PROBLEM_DETECTOR_RELEASE_PATH:-https://storage.googleapis.com/kubernetes-release}"
-  download-or-bust "${npd_hash}" "${npd_release_path}/node-problem-detector/${npd_tar}"
+  local -r npd_release_path="${NODE_PROBLEM_DETECTOR_RELEASE_PATH:-"https://github.com/kubernetes/node-problem-detector/releases/download/${npd_version}/node-problem-detector"}"
+  download-or-bust "${npd_hash}" "${npd_release_path}/${npd_tar}"
```
Comment (BenTheElder) [inline cluster/gce/gci/configure.sh:300]: we should manually check job logs
/hold
since CI clearly doesn't catch if this has a typo

**Review by wangzhen127 — COMMENTED** [review]
/test pull-kubernetes-node-e2e-containerd-standalone-mode

File: cluster/gce/gci/configure.sh (hunk 2 above)
Comment (wangzhen127) [reply cluster/gce/gci/configure.sh:300]: pull-kubernetes-node-e2e-containerd-standalone-mode does not collect node installation log and npd log. We may need to update https://github.com/kubernetes/kubernetes/blob/master/test/e2e_node/services/logs.go#L65 to get logs...

Comment (BenTheElder) [reply cluster/gce/gci/configure.sh:300]: Thanks for the pointer, yeah I don't see anything relevant in the artifacts, we'll have to patch in some confirmation

Comment (wangzhen127) [reply cluster/gce/gci/configure.sh:300]: Figured it out that we have https://testgrid.k8s.io/sig-node-node-problem-detector#ci-npd-e2e-kubernetes-gce-gci to cover the standalone mode. That CI job is failing now due to some other reason. I will fix that and prepare another PR for you to review to fix this. @BenTheElder

**Comment by wangzhen127** [comment]
The issue has been fixed by https://github.com/kubernetes/kubernetes/pull/123847. We can discard this PR now.

/close

**Pull request closed without merging**